src/test/scala/is/hail/methods/ExprSuite.scala

```diff
@@ -585,6 +588,56 @@ class ExprSuite extends SparkSuite {
     // FIXME: parser should accept minimum Long/Int literals
     // assert(eval[Long](Long.MinValue.toString+"L").contains(Long.MinValue))
     // assert(eval[Long](Long.MinValue.toString+"l").contains(Long.MinValue))
+
+    {
+      val x = eval[Map[String,IndexedSeq[Int]]]("[1,2,3,4,5].groupBy(k => if (k % 2 == 0) \"even\" else \"odd\")")
+      assert(x.isDefined)
+      x.get.keySet should contain theSameElementsAs Seq("even", "odd")
```

Comment:
I won't fight this, but I have to say I this this whole let's-write-tests-as-if-English things is horse shit.  Let's give it a fancy name some people will confuse it with a serious development methodology!  Yeah!  This is me not fighting.

I do admit our testing error messages are shitty, and if this fixes that, it is an improvement.